### ISSUE TITLE
Ship connections returning errors from Coriolis and E:D Shipyard

### ISSUE THREAD

**Issue by aadler** [Marginal/EDMarketConnector#272]
Starting recently, every time I try to click on the ship link in EDMC to get the Coriolis page, I'm getting an error. The error is below. I thought it would make sense to open an issue here before going to EDCD/coriolis. 

```
Browser: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36 Edge/15.15063
Path: 
Error: Unknown
Details:
scriptUrl: /import?data=H4sIABwDJ1oC_-VaTW_bSBL9K4GAvSVEf3_4Fltx7IGyI5jG5Ei0pbZEmCKFJumsNvB_32pSltWyZFJaT3IYwweaVle9Lr5XVV3Uz8G9s3Zwdm-y0n4cpNPBGf44yEydT-bnZlUOzn4O4qyokEzi9L-WNn_Xd80tf50VZlrU1eBsUFbGVdYNPj7f-7dZgOFBvLmftzcqW1Z39Wy2Gjw9PX0cLIppndnG0We3KGrnr9qb_uo1PKIQ05ShjcHxqpoXedKuTr46M7UY3BX54KxyNSxburRwabVq9vZoshpWIe97aCcmw90OhZCSqY3DZlly6x0l5654aHbX1x35te7oL3N36WB9PE_vq6FLHxtv3wv3cJ2PXTFztiwTgJHepxNTpUXePPA_6-o-rao0n12mNpverpY2uYyHV9ZUN6ZqTEzTcpmZ1V-to8EnEjH1r8GL50hLThhHkjIqhQYcB4z-uazShcm-mbLcY5eoSIotuzgiSgpCITqESXLA7HVe2VkbjNdIRaRpiJQiiST2v0iIAyYP4PtESKRVCJBoqRmnBAl02Np_Lmubja37o14s99kFOyxE6fdNOBUSK8UP2B0XP6wbOvNjr0kaUR1AxZQqRRVWWjFPFZvP0tzaRurP19dARooQxObl_-sUcmmzdAJR_nBpXGkbPjo7SZd2ZB9tBsue_37-fDxMRkU-uzH5zA42OeZgOhEME7LhPzzU5Ao26aaexk3a48lFBs-FHxAC2myVY0Q1p40cIO63Jn_okc4EoxwF_p8Xb3unnTJUFOlGiiPjZvbKuOmySPMKnyrFoVmY2V4V8kiRkDUMSyWoxp486gBrWntAxthOinz6jobTsnLpXV0Vh0hJWYR1YFhwhZRUWkqOhDxB4CyiKGA5YsBfAulCYcROEA5mEeMBRsU5EwQIxZiiByzezq2DvDaCorsPJIlwqG8uFdcUYEKCQ7RLjeq1GsHhh-EPm2XdQvxuzRIq85d7oFpq86qHFplmSGy0cLWsknENHxiZ0rrz2pVV8jVd3EGBajjeKUhCNRV7NEFO1UTbZYxtbivXfHIvM_guM5gWQDNI2QeZcVBswFy5Yw0LrRAGc4KeLLXTzHYKjWBgbUA4kANsHbKcwATh48oeGNutThjYyzCjgkh8ishExFAoMiAIFpJoJhmSp6mMRTo0yqESECKpIlhR_E9VGe3VyzOBAyjf6qxKL0yew6aOgsE0PMcm2KP03sb1clm4qlf9xZwF9XdrfVOCWVuCSXckGMbKI_hm0vxL80jLXgiQxgGCdm3jXLTOWff-uWJeYt69nab14h1agE7Bf0IRC7s9hPyDgK6UQL47ss3FCqQT9o5QBKkkkJ2YYIfb3HRRL9p-b0_xV5EKm3toRwWSWipFkDohi4DewySnOeIYEBLowLp73OP0Tvbr_ZgO1-ud99F7y5tupkHDzPfw7G8tqzjiNCQa5hRjAgcLoeWxRJORCIs0llwLxkGKB489XTzTEdnhmSaCQBPnH4A-5Sy1U1h8ldawbdAWRZyfVKwoVLzQJFQpqZDmDJrCfw51x5Chqz-Ku-70LKmW9KVAPC9M1oOnUb1I86IukxhSJYQfkX4Dk3FmgO7GrT4vl64wk3lcp-3Qo2NeIwnFMqgWranQThcGtg5DQ7-XNH-qgtfF7sIszWT_gQmTSJNQckT79oh7_h0i89rujZ3MmyZgj0jU7tkOrEL5gYrBND2Y398c3mA4IIZzJqUp1pRL4tMEPUHLdLfxhCMBWMIYUSIPjpjiVVnZxfuHdW33_cPaSvz9Aa_t_j-A_9bUtquj5CqdzTdR6DOEIjhsAV9Z9M2g7NsMUqWBVmgjcZ8i-jXDELLXOJZ--TaC7nEUUwwSZ1NRbszUuF7OCQs78djmZeHK7Ua4RxdOwEzThcXzdHk9RD1yPFeQ5Deu_UOFLVcglMZCMizqu8yOfE_-1dlVvwzfrsW_0bu31HP3WweQF_9jZyYVpP4M4eT7vE9ZCVzj3-Dav5rCLV_6zV4ZDRh3AemluDGTh23Odb9QotDoabxBQI5AAC
line: 0
col: 0
error: {}
```

**Comment by Marginal** [comment]
Which browser?

**Comment by aadler** [comment]
Defaults to Edge but the URL in Edge returned an error in Firefox. Can Firefox be set as the default browser in EDMC?

**Comment by Marginal** [comment]
This app uses whatever browser you have set as default in Windows. But IE and Edge can’t handle the long URLs - https://github.com/Marginal/EDMarketConnector/wiki/Troubleshooting#import-failed-no-ship-loadout-found

**Comment by aadler** [comment]
I switched the default browser to Firefox; problem solved. Thank you. Do you think that it would be possible to make the browser selection a choice in EDMC, or is that not worth the problem?

**Comment by Marginal** [comment]
> Do you think that it would be possible to make the browser selection a choice in EDMC

Yes, that’s possible. But I’m not sure how to enumerate available browsers on Windows.

**Comment by aadler** [comment]
Would this be of help?

https://stackoverflow.com/questions/9841515/list-all-web-browsers-installed-on-a-windows-machine

**Comment by tonypiz** [comment]
I'm having the same problem after the last update to EDMarketConnector. My browser is Opera and I had never experienced any problem whatsoever before when using this function. I tried setting Chrome as the default browser, but nothing changed. The error is the following:

```
Browser: Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.186 Safari/537.36 OPR/51.0.2830.55
Path:
Error: Unknown
Details:
scriptUrl: /import?data=H4sIACUbrVoC_-1abW_bOBL-K4GB270DEoHvorKfEqdpupe0uThoPwqMRdtEZUmnl3R9Rf_7jSS_SbYsxk2wh8P6Q-Ao4cwj8nmGM0N-H0xSrQfnExVm-nRggsG5PB2EqojGs0u1yAbn3wejMM6R64_MfzSvfi-eqkfl9zBWQVzkg_PBJC7SwenqwUc1B6uDv7_XoQl--Xg5uv_t_T_gz1H9fKIDnarcxJE_MdNZrtPBjx-ntSev8sS6PWW5SssRbWej9fOll1xn-VMxnS7AOFifx0ER6uqVLtJ5CRe-1Q_LbzsTgYlEzKOCrQ1eRGocR4Hy6_H--1QFGoPDOBqc52kBA5PUxKnJFzD8dPCswgLGodL7Zfz0FGpE-526lHBM1k7rgT68XTVfQzAfhybzv5hU27m-0mMV4n7HQrgu99aOq2H-uz-SME516r8LTW7p8DoFC6OZmeRXqXmu_H2J068fovs0nqY6y3wAYiZmXL1StSKfinxi8txE02ujw-BxkWj_enT1KcnNXIV3Kqv-KzBZEqrF59rXgEmH8r8N1s6xwyTlAGCftQ9Rrqc13B1DZ5g7CG1ZQo7sstOB5Yyipgns0A4L9_E3nV6l6psNEuxgXs6pjqYm0roi7er7B1g3ihBGaPP3pRiuQXhjeNmTa5VmulJFqscm0bf6WYeDc776ffX_oyv_No6mDyqa6sFaLZ3CEAwTd00VmFv_Bl4tDcr1rgQs_GEIM8U7GIM2L0gZkq50K-IUOnxU0VcLZQpGOWoAWA2uI1XtnvYSVmKEaen7ppjqG5UGSWyiHB_L2Yv5PB6GJikx7Fte0eaZxzpYcqXmarrPBvUc5jWJ5jHvoJV7nY50Gbp-2pzJ8tQ8FXncSWCyowPSYexxplMQ9y1EcYupwg4SfUKQu0IAJydX33QY7mqAtDTwRasEIuynZ52OZyqd6qBfCEJiz-NrHt4kuX9fwD_cqgxi5nszf4IYWlKrVweScyEQOMwSAAUxr0235XP_Ww1TT-CvRkewJUZFGJZTc1uC_nkOdzIPU4fQZmyihB7NvJeZ62Uelg5mzRiOpTgiBmPmeG5rM0BAr_UHH0VoRh3hNcxyQZG39emhNxF76B0HJ7-WHL-EcAfZz68nd-N_FSaK2mSn-8n-bs2i_pjPPCZwg-p3RZiboYoisLTkesXBXrIz16PShuwmGusoMCpd-GlcREHWxXby-myHdXeb9ISN6ni2v8hcP9tBPLJJUuGSI9guHdnakfBxAZvB0Ba_vTI32ZLNG_Kb_J_zm74-vyFdxk1CYoaP5_eLzNnw2xUtfhN2TDRHjmgJxaNHMhw7kjYZLrH7F8OPZriZ6FGRJHGaW2X8mMtGxr81fjvp76rIyRo8EZRWyeSdMtG7aj0yKwTIEw0E9djKuVs7Zx3O6UYYDAspcLXb3-nAFPNmytbfkpBeV8L5WKSphlWtzPaXXsRjEu3BQaxwAJIGjkuVmfGdyTIT6gc1_upfmz90cBjMZk0Y5NFeDcaEJge-IGxXBxLSWJPRrJTzEFL-y1Y1aFGMEoyIrArue6BG_nv8ZNOw8Vy2qUVXA_11u-i2mJsoLjJ_BEEPogwidn2U-1BFupyJiyRJYzWejYqyB9PfyHEJxc3qvDbVtNOHgWFWoaiC6yZgH7sRLXU2VIkad7Rh6E513JUgLY096LpS22OM0GbchzqUCmmRxY8WEIbnr4RzaewtcNbx_5VwLo39HM5Xrc7bBUubhv6Nmc6uU_3vQkfjhVXLirRF0TZZxgppGysI4ZJ4dKORUmN2GxkSYhdIUg7fhsD6Fco8zDiuemcPKlBHa_NwcxTt0OhlvVHsOazVXUVdhflIR1mcPpYkzEdjFV1E03BvJ22nY-v19knJLgFvtTkZznQRTdv8wy3-1cAg55jO8i-6_GnHOc6bu1NlJtteZ9pLNSowJW4lMNjdkhIRstiXuIR8b-29HAUzmOt1Tx9h_8vMuqtfHcXgOtex25gZa7z6EBY1rhKDrXyp__gEygjpiQ0EUveXLSHQAxCELQTKKSfuGgHdIDhqIyynPzPP-kFnEHwUJMh7CM53VPfWJxv_BGHkZnwQFdtBJV4WC4h0ULNL4GIPW2x3h2o8IhzaMgqS6YowVX74XkeHD5fO2E6AEX0WIb-D-JLP9pgTzXK27Ga74lAV2rMQe_vsrxn_2vvv-iXLk1sIhNXsmcymSV6GQenuSdI31l52ZMRdlyNGLarPrPLjp9VUVmeRq-KzUjL702MJ3xysWyDABxAwWwREQt6wBiBsAQj4yN0jt9E4jpOXVVlSMiE5WUNwbSG4xPVoC0J9f-BSLWxaABsMXGCyBUFubh30zoKLm1v69eiqDMFpYMYrKjPbnR0KNcyZ2Hf7obfGI16TDpflXYfVRCwRkF4ELhaIrfxjZD8PlIum_1EOibxKL-NgAUn1OH7W6aJM4UDj_kXwXAayoH-350hQl68B1XstsdMHlQf0gW31IYms3MPeEL7hadoLNhiLU1yEIQHesnZGHIQ9jqFGAa64kh3diSXEEc1OrOvSLnO_w4N689nJ13f2q6485AFy1E-T6_J-y54XhWqEus0XxYgKQSjBnHoMd5W4D7q8L_Ro5nuzLeLI5uEhA2G8bkXL9jdwH1RiguptbRq4VPadNle87Y9_zLXp3iYzlcEs-lldZ2_vnw15kP9teRBPNkl8BnKRgnvVKjIsCDr-KI6ALdIUiERu42D3zxILEo7HWlFBeBQCAkaEYyElPUot1CG4qRbhen-ppVstjyZavLDNz3GzzX-jVT4y0dfb6nbmptlfmu7PM7y6P9fAQd62gmUOd1sdGkaPqWHtI4FVCfsCWAev6u3c1EP8Jwq6_bDetKNV1UaXcVweO_oP6xKp_9ITZO9UNMjZtFWmW8i6j0ohsfd2yEltclDCafMMaphCxFolnVswWH-PjQvOxQ4MZgdj6zJkBSOMi2APjP6CAGO5LI0-gxx1-Dk2Y4sjHwnJ71Z3uxp1X2a-Qx3qzLbFV8fYYRxa3Y52IePyNi2-5eAiy-O5yerbyvdFmoRWzn_s3LKGUVC9Jx-u4Fk0P3PZ8sGSvR91HJ3My80Z5mmjm--DGYTcch5d7iHpiq2L35gygTxMYZ7yOFflmTonpOw5AYmKCDa8CCqUc5chDyLMj_8ChWebRB0vAAA%3D
line: 0
col: 0
error: {}
```

**Comment by Marginal** [comment]
@tonypiz Please see EDCD/coriolis#205

**Comment by Noesis-** [comment]
> > Do you think that it would be possible to make the browser selection a choice in EDMC

> Yes, that’s possible. But I’m not sure how to enumerate available browsers on Windows.

Hi @Marginal I'm just thinking out loud here but I'm inclined to think rather than enumerating available browsers, allowing a user defined alternate browser would actually be better (i.e. a "drive:\path\browser.exe" string).  Reason being enumeration would pick up installed browsers yet miss any portable (non-installed) browsers.

I'm basing this on the assumption that I'm not the only person who uses a different (portable) browser when gaming, than my main default browser, and additionally it would be easier to implement, i.e assume if the string is blank, default browser is used like now, otherwise the string is used as the browser instead.

**Comment by aadler** [comment]
Hasnb't been a problem for a while. Closing.
